drivers/wifi/esp32/src/esp_wifi_drv.c

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2020 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT espressif_esp32_wifi
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(esp32_wifi, CONFIG_WIFI_LOG_LEVEL);
+
+#include <net/ethernet.h>
+#include <net/net_pkt.h>
+#include <net/net_if.h>
+#include <device.h>
+#include <soc.h>
+#include <ethernet/eth_stats.h>
+#include "esp_networking_priv.h"
+#include "esp_private/wifi.h"
+#include "esp_event.h"
+#include "esp_timer.h"
+#include "esp_wifi_system.h"
+#include "esp_wpa.h"
+
+#define DEV_DATA(dev) \
+	((struct esp32_wifi_runtime *)(dev)->data)
+
+struct esp32_wifi_runtime {
+	struct net_if *iface;
+	uint8_t mac_addr[6];
+	bool tx_err;
+	uint32_t tx_word;
+	int tx_pos;
+	uint8_t frame_buf[NET_ETH_MAX_FRAME_SIZE];
+#if defined(CONFIG_NET_STATISTICS_ETHERNET)
+	struct net_stats_eth stats;
+#endif
+};
+
+static int eth_esp32_send(const struct device *dev, struct net_pkt *pkt)
+{
+	const int pkt_len = net_pkt_get_len(pkt);
+
+	/* Read the packet payload */
+	if (net_pkt_read(pkt, DEV_DATA(dev)->frame_buf, pkt_len) < 0) {
+		return -EIO;
+	}
+
+	/* Enqueue packet for transmission */
+	esp_wifi_internal_tx(ESP_IF_WIFI_STA, (void *)DEV_DATA(dev)->frame_buf, pkt_len);
+
+	LOG_DBG("pkt sent %p len %d", pkt, pkt_len);
+
+	return 0;
+}
+
+esp_err_t eth_esp32_rx(void *buffer, uint16_t len, void *eb)
+{
+	struct net_pkt *pkt;
+	struct net_if *iface;
+
+	iface = net_if_get_first_by_type(&NET_L2_GET_NAME(ETHERNET));
```

Comment:
This will prevent using ESP with any other ethernet network driver. It would be better to save iface pointer globally in this driver, e.g. during driver or network interface initialization.

Reply:
PTAL.